lib/archethic/transaction_chain/transaction/validation_stamp/ledger_operations.ex

```diff
@@ -108,13 +131,28 @@ defmodule Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperation
     |> ledger_balances(%{uco: fee, nft: %{}})
   end
 
-  defp ledger_balances(movements, acc \\ %{uco: 0, nft: %{}}) do
+  defp ledger_balances(movements, acc) do
     Enum.reduce(movements, acc, fn
       %{type: :UCO, amount: amount}, acc ->
         Map.update!(acc, :uco, &(&1 + amount))
 
-      %{type: {:NFT, nft_address}, amount: amount}, acc ->
-        update_in(acc, [:nft, Access.key(nft_address, 0)], &(&1 + amount))
+      %{to: to, type: {:NFT, nft_address, nft_id}, amount: _amount}, acc ->
+        # update_in(acc, [:nft, Access.key(nft_address, 0)], &(&1 + amount))
+        put_in(acc, [:nft, nft_address], {to, nft_id})
+
+      %{type: :call}, acc ->
+        acc
+    end)
+  end
+
+  defp utxo_ledger_balances(movements, acc \\ %{uco: 0, nft: %{}}) do
+    Enum.reduce(movements, acc, fn
+      %{type: :UCO, amount: amount}, acc ->
+        Map.update!(acc, :uco, &(&1 + amount))
+
+      %{from: from, type: {:NFT, nft_address, nft_id}, amount: _amount}, acc ->
+        # update_in(acc, [:nft, Access.key(nft_address, 0)], &(&1 + amount))
+        put_in(acc, [:nft, nft_address], {from, nft_id})
```

Comment:
This changes only works for single NFT utxo but not for fungible tokens based.
As we get multiple time the same token as input.
It should also return balances